packages/venia-ui/lib/RootComponents/Category/categoryContent.js

```diff
@@ -58,7 +71,7 @@ const CategoryContent = props => {
                 <div className={classes.pagination}>
                     <Pagination pageControl={pageControl} />
                 </div>
-                {modal}
+                <Suspense fallback={null}>{modal}</Suspense>
```

Comment:
On a slow network this can cause a weird UX where we display the overlay until the modal component is fully loaded and then we render it in. In the below gif you can see I hover/click quickly and but the chunks are slow to load and so it looks off.

I would suggest that we might create some sort of skeleton drawer/modal that we can render until then?


[![Image from Gyazo](https://i.gyazo.com/fb22c6a706bcca509c7027fabd32172d.gif)](https://gyazo.com/fb22c6a706bcca509c7027fabd32172d)

Reply:
Great call. I actually did go down this path but found it to be much more more than anticipated because the `FilterModal` itself contains all the styles for rendering the drawer / modal (white background, animation in, etc.).

We could duplicate them but that could cause maintenance headaches keeping them in sync.
I also tried lazy loading the _contents_ of the modal, but by that time we've already eaten most of the cost and I didn't think it was worth it.

It would be much easier to do if we had `LeftDrawer` / `RightDrawer` components that we completely replaced the children of.

Right now we're injecting into new modals / drawers each time:

**Before "Filter" Button Click**:
![Screen Shot 2019-11-14 at 3 02 44 PM](https://user-images.githubusercontent.com/13182778/68896039-3d626e00-06f0-11ea-804e-47814366d20e.png)

**After "Filter" Button Click**:
![Screen Shot 2019-11-14 at 3 02 57 PM](https://user-images.githubusercontent.com/13182778/68896051-43584f00-06f0-11ea-9084-022f40b887e6.png)

I agree the experience is not good - I'm in the middle of duplicating the minimum set of "modal" styles now, as re-writing the drawer system is too large / out of scope for this imo.

Reply:
A drawer placeholder that can slide in from left or right would be a good first step!

Reply:
Okay, https://github.com/magento/pwa-studio/pull/1971/commits/a0228590afc1452be253e23f3681dfbb362f38ed loads the filters immediately so supporting the drawer placeholder / skeleton becomes less of an issue right now.

I'll create a separate ticket for re-working the drawers.